phpunit/class-rest-scripts-controller-test.php

```diff
@@ -0,0 +1,287 @@
+
+	/**
+	 * Test whether proper routes are registered.
+	 */
+	public function test_register_routes() {
+		$routes = rest_get_server()->get_routes();
+		$this->assertArrayHasKey( '/__experimental/scripts', $routes );
+		$this->assertArrayHasKey( '/__experimental/scripts/(?P<handle>[\w-]+)', $routes );
+	}
+
+	/**
+	 * Test context param.
+	 */
+	public function test_context_param() {
+		$request  = new WP_REST_Request( 'OPTIONS', '/__experimental/scripts' );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertEquals( 'view', $data['endpoints'][0]['args']['context']['default'] );
+		$this->assertEquals( array( 'view', 'embed', 'edit' ), $data['endpoints'][0]['args']['context']['enum'] );
+		// Single.
+		$request  = new WP_REST_Request( 'OPTIONS', '/__experimental/scripts/' . self::$script_handle );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertEquals( 'view', $data['endpoints'][0]['args']['context']['default'] );
+		$this->assertEquals( array( 'view', 'embed', 'edit' ), $data['endpoints'][0]['args']['context']['enum'] );
+	}
+
+	/**
+	 * Test multiple scripts.
+	 */
+	public function test_get_items() {
+		wp_set_current_user( self::$admin_id );
+		$request = new WP_REST_Request( 'GET', '/__experimental/scripts' );
+		$request->set_query_params( array( 'dependency' => 'script1' ) );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertCount( 0, $data );
+	}
+
+	/**
+	 * Test multiple scripts with nested dependencies.
+	 */
+	public function test_get_items_nested_deps1() {
+		wp_set_current_user( self::$admin_id );
+		$request = new WP_REST_Request( 'GET', '/__experimental/scripts' );
+		$request->set_query_params( array( 'dependency' => 'script-with-deps' ) );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertCount( 2, $data );
+
+		$this->assertEquals( 'dependency1', $data[0]['handle'] );
+		$this->assertEquals( 'dependency2', $data[1]['handle'] );
+
+		$this->assertCount( 0, $data[0]['deps'] );
+		$this->assertCount( 0, $data[1]['deps'] );
+	}
+
+	/**
+	 * Test multiple scripts with nested dependencies.
+	 */
+	public function test_get_items_nested_deps2() {
+		wp_set_current_user( self::$admin_id );
+		$request = new WP_REST_Request( 'GET', '/__experimental/scripts' );
+		$request->set_query_params( array( 'dependency' => 'script-with-nested-deps' ) );
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+		$this->assertCount( 3, $data );
+
+		$this->assertEquals( 'dependency3', $data[0]['handle'] );
+		$this->assertEquals( 'dependency4', $data[1]['handle'] );
+		$this->assertEquals( 'dependency5', $data[2]['handle'] );
+
+		for ( $i = 0; $i < 3; $i ++ ) {
+			$keys = array( 'src', 'url', 'args', 'ver', 'extra', 'textdomain', 'translations_path', 'deps', '_links' );
```

Comment:
Dont define key in loop.

Reply:
🤦

Reply:
Fixed in both places. Some leftover before I ran this code in a loop but still 🤦